### ISSUE TITLE
Polybar doesn't show hours on clock and doesn't take entire screen width

### ISSUE THREAD

**Issue by RossComputerGuy** [polybar/polybar#1682]
## Describe the issue
*A clear and concise description of what the bug is.*
The hours aren't showing and the tray is offsetted.

**Expected behavior:**

The bar to take the entire screen width.

**Actual behavior:**
The hours aren't showing and the tray is offsetted.

**Was it working before?**
* Did you also experience this bug in an earlier version of polybar? No
* If no, what was the last version where this worked correctly? 3.3.0

## To Reproduce
*A minimal but complete config with which the problem occurs:*
```dosini
[bar/top]
;monitor = ${env:MONITOR:HDMI-1}
width = 100%
height = 27
fixed-center = false

background = ${colors.background}
foreground = ${colors.foreground}

line-size = 3
line-color = #f00

border-size = 0
border-color = #00000000

padding-left = 1
padding-right = 1

module-margin-left = 1
module-margin-right = 1

modules-right = data
tray-position = right
tray-padding = 2

[module/date]
type = internal/date
interval = 5

time = %H:%M
time-alt = %H:%M:%S

format-prefix-foreground = ${colors.foreground-alt}
format-underline = ${colors.underline}

label = %time%
```
*List any other steps needed to reproduce the issue besides starting polybar with the config you posted above.*

## Polybar Log
*Post everything polybar outputs to the terminal when you run it and the issue occurs below*
```
warn: No monitor specified, using "LVDS-0"
error: module/xbacklight: Could not get data (err: XCB_NAME (15))
error: Disabling module "xbacklight" (reason: Not supported for "LVDS-0")
```

## Screenshots
![screenshot from 2019-03-08 07-45-24](https://user-images.githubusercontent.com/19699320/54039083-e24aa580-4176-11e9-996b-a4f6919e4e3b.png)

## Environment:
* WM: Cinnamon
* Output of `polybar -vvv`:
```
polybar 3.3.0

Features: +alsa +curl -i3 -mpd +network(libnl) +pulseaudio +xkeyboard

X extensions: +randr (+monitors) -render -damage -sync +composite +xkb +xrm +xcursor

Build type: RelWithDebInfo
Compiler: /usr/bin/c++
Compiler flags: -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -Wall -Wextra -Werror -Wno-noexcept-type -pedantic -pedantic-errors
Linker flags: -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now
```

## Additional context
*Add any other context that you think is necessary about the problem here.*

**Comment by patrick96** [comment]
Wow, sorry this is so late, kinda got flodded the past few months.

---

>  Did you also experience this bug in an earlier version of polybar? No
    If no, what was the last version where this worked correctly? 3.3.0

So this means that this both worked and didn't work on the same version?

Does this happen reliably even after restarting polybar?

We are constantly having issues with the tray, especially with the tray overlapping bar elements (see #591). But the issue where the tray is actually misplaced isn't much reported. @Yves1024 had the same issue but it was fixed after restarting X. Others have set `fixed-center = false` in the bar section which fixed it.

Are you using a high DPI screen? Some are reporting issues only when using high DPI screens.

**Comment by SpaceboyRoss01** [comment]
Yeah, the issue randomly occurs. If I run `polybar-msg cmd toggle`, it doesn't fix the bug but if I kill the process and respawn it then it'll work. I don't think the Dell Latitude E6420 and ASUS X401A screens are high DPI.

**Comment by patrick96** [comment]
Does `fixed-center = false` work for you?

I just checked, no they are not even 1080 screens.

**Comment by SpaceboyRoss01** [comment]
Since the bug occurs at random, I dunno if `fixed-center` even fixes it. My Dell Latitude E6420 reports the screen is 1366x768.

**Comment by patrick96** [comment]
That's true. That's also the main problem why we cannot really fix it in polybar because it's almost impossible to reliably reproduce. You'll have to wait and see if it comes back.

**Comment by patrick96** [comment]
Hi, we have recently fixed #591 and it may also fix your issue. If you have version 3.4.2 or newer, do you still see this issue?
